web/src/settings_invites.js

```diff
@@ -214,6 +261,70 @@ export function on_load_success(invites_data, initialize_event_handlers) {
         $(".dialog_submit_button").attr("data-is-multiuse", meta.is_multiuse);
     });
 
+    $(".admin_invites_table").on("click", ".edit-invite", (e) => {
+        // This click event must not get propagated to parent container otherwise the modal
+        // will not show up because of a call to `close_active_modal` in `settings.js`.
+        e.stopPropagation();
+        e.preventDefault();
+
+        meta.invite_id = Number.parseInt($(e.currentTarget).attr("data-invite-id"), 10);
+        const html_body = render_edit_invite_user_modal({
+            invite_as_options: settings_config.user_role_values,
+            is_admin: page_params.is_admin,
+            is_owner: page_params.is_owner,
+            streams: get_invite_streams(),
+            notifications_stream: stream_data.get_notifications_stream(),
+        });
+
+        function invite_user_modal_post_render() {
+            $("#edit-invite-form .dialog_submit_button").prop("disabled", true);
+            const initial_invite = meta.invites.find(
+                (invite) => invite.id === meta.invite_id && invite.is_multiuse,
+            );
+            for (const stream_id of initial_invite.streams) {
+                $(`[value=${stream_id}]`).prop("checked", true);
+            }
+            $("#invite_as").val(initial_invite.invited_as);
+
+            function state_unchanged() {
+                const initial_streams = initial_invite.streams.sort();
+                let selected_streams = [];
+                $("#streams_to_add input:checked").each((_, stream) => {
+                    const stream_id = Number.parseInt(stream.value, 10);
+                    selected_streams.push(stream_id);
+                });
+                selected_streams = selected_streams.sort();
+                return (
+                    selected_streams.length === initial_streams.length &&
+                    selected_streams.every((val, index) => val === initial_streams[index]) &&
+                    Number.parseInt($("#invite_as").val(), 10) === initial_invite.invited_as
+                );
+            }
+
+            $("#edit-invite-form").on("change", "input:checkbox, select", () => {
+                $("#edit-invite-form .dialog_submit_button").prop("disabled", state_unchanged());
+            });
+            $("#invite_check_all_button").on("click", () => {
+                $("#streams_to_add :checkbox").prop("checked", true);
+                $("#edit-invite-form .dialog_submit_button").prop("disabled", state_unchanged());
+            });
+            $("#invite_uncheck_all_button").on("click", () => {
+                $("#streams_to_add :checkbox").prop("checked", false);
+                $("#edit-invite-form .dialog_submit_button").prop("disabled", state_unchanged());
```

Comment:
Just noting here that there is a PR opened recently to add this functionality of enabling/disabling the button for all modals. #24911 is the PR. Nothing to change for now, but we can update this if #24911 gets merged first.